mode/soy/soy.js

```diff
@@ -266,6 +266,17 @@
               state.quoteKind = match;
               return "string";
             }
+            if (match = stream.match(/(true|false)(?!\w)/)) {
+              return "atom";
+            }
+            if (match = stream.match(/0x([0-9a-fA-F]{2,})/)) {
+              // Match hexadecimal number.
+              return "atom";
+            }
+            if (match = stream.match(/-?([0-9]*[.])?[0-9]+/)) {
+              // Match all decimals and whole numbers.
+              return "atom";
+            }
```

Comment:
You can put all of these in a single `if` with `||`, and you don't need to assign to `match` here since you aren't going to read from it. Could you make those changes?